resources/images/filters.go

```diff
@@ -174,6 +176,66 @@ func (*Filters) Padding(args ...any) gift.Filter {
 	}
 }
 
+// Dither creates a filter that dithers an image.
+func (*Filters) Dither(options ...any) gift.Filter {
+	ditherOptions := struct {
+		colors     []string
+		method     string
+		serpentine bool
+		strength   float32
+	}{
+		colors:     []string{"000000ff", "ffffffff"},
+		method:     "floydsteinberg",
+		serpentine: true,
+		strength:   1.0,
+	}
+
+	var opt maps.Params
+	if len(options) > 0 {
+		opt = maps.MustToParamsAndPrepare(options[0])
```

Comment:
I was going to hit the merge button, but I'm going to request one more improvement.

In most similar places, we use `mapstructure.WeakDecode` to read from a map into a struct. That works great in most simple cases (and is what we use to read the main config).

I suspect you need to make the struct fields exported (first upper char) to make this work.

This is nit-picking, but I have spent a lot of time cleaning up old code trying, when possible, to do it one way, me may as well continue that "new year's resolution".

Note that I don't mind the current `switch` approach in situations where mapstructure doesn't work/is too cumbersome, but I don't think this is the case.